aimmdb/ingest/eli.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+
+def read_metadata_and_header(path: Path):
+    metadata = {}
+    header = ""
+    dat_file = path.open()
+    for line in dat_file:
+        if line.startswith("#"):
+            line = line[2:]  # remove hash and whitespace
+            if ":" in line:
+                key, value = line.split(":", maxsplit=1)
+                metadata[key] = value.strip()
+            else:
+                header = line
+    return metadata, header
+
+
+def ingest(path: Path, return_uid=False):
+    md, hdr = read_metadata_and_header(path)
+    # print(header)
+    temp_data = pd.read_csv(path, sep="\s+", comment="#", names=hdr.split())
+    temp_data["mu_trans"] = -np.log(temp_data["it"] / temp_data["i0"])
+    temp_data["mu_fluor"] = -(temp_data["iff"] / temp_data["i0"])
+    temp_data["mu_ref"] = -np.log(temp_data["ir"] / temp_data["i0"])
+
+    data = temp_data[["energy", "mu_trans", "mu_fluor", "mu_ref"]]
+
+    if return_uid:
+        uid = md["Scan.uid"]
+        return data, md, uid
+    else:
+        return data, md
+
+
+if __name__ == "__main__":
+
+    def test():
```

Comment:
We should definitely test the code but the tests should be in a separate directory. I prefer `aimmdb/_tests/ingest/test_eli.py` for this. Try to structure them in a way such that they can run automatically on any machine independent of any particular path. It also might make sense to store a dummy data file in the repository here (with Eli's permission).